core/src/test/java/org/elasticsearch/deps/joda/SimpleJodaTests.java

```diff
@@ -260,6 +260,10 @@ public void testThatEpochsCanBeParsed() {
         } else {
             assertThat(dateTime.getMillisOfSecond(), is(0));
         }
+
+        // test floats get truncated
+        String epochFloatValue = String.format(Locale.US, "%d.%d", dateTime.getMillis() / (parseMilliSeconds ? 1L : 1000L), randomNonNegativeLong());
+        assertThat(formatter.parser().parseDateTime(epochFloatValue).getMillis(), is(dateTime.getMillis()));
```

Comment:
I think we should accept numbers as defined by javascript/JSON schema, and not formatted string representation of numbers. The issue is that we didn't accept the numbers from javascript which does not have an integer datatype. Accepting string representation of valid javascript numbers is the bonus part since all dates are parsed as strings anyway.